### DISCUSSION TITLE
What branching strategy should we use?

### DISCUSSION THREAD

**Discussion by endrebak** [biocore-ntnu/pyranges#320] [General]
Currently, we merge PRs continually directly onto master. Is it preferable to only commit version changes to master?

I.e. so that instead of merging commit 1, 2, and 3 onto master we save all of these up on the develop branch and then merge them collectively into master when bumping the version and pushing to PyPi?

I'm most familiar with Gerrit which uses a cherry-picking strategy.

@marco-mariotti @JoanPAAL

**Comment by endrebak** [comment]
Seems like the current one (lots of small bumps on trunk) is considered the best one: https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development

**Comment by marco-mariotti** [comment]
I think that if CI is well done (as it seems!) we can dare to do PRs directly on the master.
I would recommend to go for a developer branch only if we plan major changes (e.g. the switch to unique dataframe one)
